kythe/go/platform/tools/kzip/stripcmd/stripcmd.go

```diff
@@ -0,0 +1,152 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package mergecmd provides the kzip command for merging archives.
+package stripcmd // import "kythe.io/kythe/go/platform/tools/kzip/stripcmd"
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"path/filepath"
+
+	"kythe.io/kythe/go/platform/kzip"
+	"kythe.io/kythe/go/platform/tools/kzip/flags"
+	"kythe.io/kythe/go/platform/vfs"
+	"kythe.io/kythe/go/util/cmdutil"
+
+	"bitbucket.org/creachadair/stringset"
+	"github.com/google/subcommands"
+)
+
+type stripCommand struct {
+	cmdutil.Info
+
+	input    string
+	output   string
+	append   bool
+	encoding flags.EncodingFlag
+}
+
+// New creates a new subcommand for merging kzip files.
+func New() subcommands.Command {
+	return &stripCommand{
+		Info:     cmdutil.NewInfo("strip", "strip units from kzip file", "--input path --output path unit-hash*"),
+		encoding: flags.EncodingFlag{Encoding: kzip.EncodingJSON},
+	}
+}
+
+// SetFlags implements the subcommands interface and provides command-specific flags
+// for merging kzip files.
+func (c *stripCommand) SetFlags(fs *flag.FlagSet) {
+	fs.StringVar(&c.output, "output", "", "Path to output kzip file")
+	fs.StringVar(&c.input, "input", "", "Path to input kzip file")
+	fs.Var(&c.encoding, "encoding", "Encoding to use on output, one of JSON, PROTO, or ALL")
+}
+
+// Execute implements the subcommands interface and merges the provided files.
+func (c *stripCommand) Execute(ctx context.Context, fs *flag.FlagSet, _ ...interface{}) subcommands.ExitStatus {
+	if c.output == "" {
+		return c.Fail("required --output path missing")
+	}
+	if c.input == "" {
+		return c.Fail("required --input path missing")
+	}
+	opt := kzip.WithEncoding(c.encoding.Encoding)
+	dir, file := filepath.Split(c.output)
+	if dir == "" {
+		dir = "."
+	}
+	tmpOut, err := vfs.CreateTempFile(ctx, dir, file)
+	tmpName := tmpOut.Name()
+	defer func() {
+		if tmpOut != nil {
+			tmpOut.Close()
+			vfs.Remove(ctx, tmpName)
+		}
+	}()
+	if err != nil {
+		return c.Fail("Error creating temp output: %v", err)
+	}
+	units := stringset.FromKeys(fs.Args())
+	if err := stripArchive(ctx, tmpOut, c.input, units, opt); err != nil {
+		return c.Fail("Error stripping archives: %v", err)
+	}
+	if err := vfs.Rename(ctx, tmpName, c.output); err != nil {
+		return c.Fail("Error renaming tmp to output: %v", err)
+	}
+	return subcommands.ExitSuccess
+}
+
+func stripArchive(ctx context.Context, out io.WriteCloser, input string, digests stringset.Set, opts ...kzip.WriterOption) error {
+	wr, err := kzip.NewWriteCloser(out, opts...)
```

Comment:
maybe open this up right before the scan, before it's used

(or not, if you're intentionally doing it earlier for verification)

Reply:
done